browser/js/cart/cart.factory.js

```diff
@@ -35,40 +48,85 @@ app.factory('CartFactory', function ($http, ProductFactory, localStorageService)
       cart.push(cartItem);
 
       syncLocalStorage();
-      return $http.post('/api/orders/' + orderId + '/item', cartItem)
-      .then( function(order) {
-        return order;
-      })
+      $rootScope.$broadcast('cart-updated');
+
+      // syncronize with database if logged in
+      if(isLoggedIn()) {
+        if(!orderId) { // if no pending order was found, create one
+          return createNewOrder()
+          .then(function() {
+            return $http.post('/api/orders/' + orderId + '/item', cartItem)
+          })
+          .then(res => res.data);
+        } else { // append to existing order
+          return $http.post('/api/orders/' + orderId + '/item', cartItem)
+          .then(res => res.data);
+        }
```

Comment:
your if / else statement has repeated code, would condense
i think the 'if' statement still returns undefined for orderId bc you haven't returned it as a param